docker/ssl/nginx-image/nginx-secure.conf

```diff
@@ -0,0 +1,51 @@
+events { worker_connections 1024; }
+http {
+	server {
+		listen 80;
+		server_name ___my.example.com___;
+
+		location /.well-known/acme-challenge {
+			proxy_pass http://___LETSENCRYPT_IP___:___LETSENCRYPT_PORT___;
+			proxy_set_header Host            $host;
+			proxy_set_header X-Forwarded-For $remote_addr;
+			proxy_set_header X-Forwarded-Proto https;
+		}
+
+		location / {
+			return         301 https://$server_name$request_uri;
```

Comment:
Will wallets / clients follow the redirect to the SSL version? In addition all our API calls are POST, and by default clients won't follow redirects on a POST, thus switching to the GET method, resulting in a 500 from the API.
I would say to just send the plain request to IRI. 80 for plain HTTP and 443 for HTTPS. 


```
* Rebuilt URL to: http://irissl.EDITED/
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 52.49.30.40...
* TCP_NODELAY set
* Connected to irissl.EDITED (52.49.30.40) port 80 (#0)
> POST / HTTP/1.1
> Host: irissl.EDITED
> User-Agent: curl/7.61.0
> Accept: */*
> X-IOTA-API-VERSION: 1.4.1.2
> Content-Length: 28
> Content-Type: application/x-www-form-urlencoded
> 
} [28 bytes data]
* upload completely sent off: 28 out of 28 bytes
< HTTP/1.1 301 Moved Permanently
< Server: nginx/1.14.0
< Date: Mon, 17 Sep 2018 16:31:31 GMT
< Content-Type: text/html
< Content-Length: 185
< Connection: keep-alive
< Location: https://irissl.EDITED/
< 
* Ignoring the response-body
{ [185 bytes data]

100   213  100   185  100    28   1697    256 --:--:-- --:--:-- --:--:--  1972
* Connection #0 to host irissl.EDITED left intact
* Issue another request to this URL: 'https://irissl.EDITED/'
* Switch from POST to GET
*   Trying 52.49.30.40...
* TCP_NODELAY set
* Connected to irissl.EDITED (52.49.30.40) port 443 (#1)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: none
  CApath: /etc/ssl/certs
} [5 bytes data]
* (304) (OUT), TLS handshake, Client hello (1):
} [512 bytes data]
* (304) (IN), TLS handshake, Server hello (2):
{ [108 bytes data]
* TLSv1.2 (IN), TLS handshake, Certificate (11):
{ [2737 bytes data]
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
{ [300 bytes data]
* TLSv1.2 (IN), TLS handshake, Server finished (14):
{ [4 bytes data]
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
} [37 bytes data]
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
} [1 bytes data]
* TLSv1.2 (OUT), TLS handshake, Finished (20):
} [16 bytes data]
* TLSv1.2 (IN), TLS handshake, Finished (20):
{ [16 bytes data]
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
* ALPN, server accepted to use http/1.1
* Server certificate:
*  subject: CN=irissl.EDITED
*  start date: Sep 17 15:05:15 2018 GMT
*  expire date: Dec 16 15:05:15 2018 GMT
*  subjectAltName: host "irissl.EDITED" matched cert's "irissl.EDITED"
*  issuer: C=US; O=Let's Encrypt; CN=Let's Encrypt Authority X3
*  SSL certificate verify ok.
} [5 bytes data]
> GET / HTTP/1.1
> Host: irissl.EDITED
> User-Agent: curl/7.61.0
> Accept: */*
> X-IOTA-API-VERSION: 1.4.1.2
> 
{ [5 bytes data]
< HTTP/1.1 500 Internal Server Error
< Server: nginx/1.14.0
< Date: Mon, 17 Sep 2018 16:31:31 GMT
< Content-Type: application/json
< Content-Length: 66
< Connection: keep-alive
< Access-Control-Allow-Origin: *
< 
{ [66 bytes data]

100    66  100    66    0     0    164      0 --:--:-- --:--:-- --:--:--   164
* Connection #1 to host irissl.EDITED left intact
{"exception":"Invalid request payload: \u0027\u0027","duration":0}
```